### PR TITLE
str(SkyCoord) should not return repr-style output

### PR DESCRIPTION
**Pull request by astrofrog**
At the moment `str(SkyCoord)` returns a repr-style output:

```
In [20]: str(c)
Out[20]: '<SkyCoord (ICRS): ra=10.68470833 deg, dec=41.26875 deg>'
```

On the other hand, `Time` returns a nicely formatted time:

```
In [21]: str(t)
Out[21]: '2015-01-20 12:46:46.735742'
```

I think the second is correct, and `str(c)` should return 'nice' output. I would suggest that `str` should return the same output as `c.to_string()`.

Furthermore, `to_string` at the moment could return a better default:

```
In [22]: c
Out[22]: <SkyCoord (ICRS): ra=10.68470833 deg, dec=41.26875 deg>

In [23]: c.to_string()
Out[23]: '10.6847 41.2687'
```

For equatorial systems the default really should be `hmsdms`:

```
In [18]: c.to_string(style='hmsdms')
Out[18]: '00h42m44.33s +41d16m07.5s'
```

So there are two requests here:
- Tie `str(SkyCoord)` to `SkyCoord.to_string`
- Have frames define a default syle for `to_string`

Both of these are trivial so I'm happy to implement them if @eteq and @taldcroft agree.

By the way, this will result in more 'normal' output for `SkyCoord` in tables following #3011

### CODE REVIEW

**Comment by mhvk** [comment]
:+1: to linking `__str__` to `to_string()`. But whatever the default is, it *should* include the sign for the declination!

On multiple coordinates: I think one string presented as a list of strings is the best solution. How about just returning `str(self.to_string())`? This is what `Time` effectively does internally as well.

p.s.   We should try to define default formats consistently through astropy. `Time` can do it, but has several knobs instead of just one (`out_fmt`, `precision`, ...).. My preference would be to define `__format__` properly in each class and then have an associated attribute that stores the default format string that is used (just like a float implicitly has `'f'` as a default format string). There was a _long_ discussion about implementing `__format__` for coordinates on astropy-dev; so far, it has been implemented in `Unit` and `Quantity` (though the latter gives only partial control). Anyway, for another PR...

**Comment by astrofrog** [comment]
One issue I found is also that the default ``to_string`` doesn't show the distance, but if this is a full part of the coordinate maybe it should be. But I don't have strong feelings.

+1 for consistency across Astropy. I like the ``format`` functionality of ``Unit`` and ``Quantity`` and we should have more of that!

**Comment by astrofrog** [comment]
I've now changed this to return ``str(SkyCoord.to_string)`` which means that it works for array output too:

File: CHANGES.rst [hunk 1]
```diff
@@ -295,6 +296,9 @@ API Changes
     class has now been deprecated, except in the case where a frame with data
     is passed as the sole positional argument. [#3152]
 
+  - ``str(SkyCoord(...))`` now returns the same as ``str(SkyCoord.to_string)``.
```
Comment (mhvk) [inline CHANGES.rst]: should be `same as `str(SkyCoord.to_string())` (add parentheses at end).

**Comment by mhvk** [comment]
@astrofrog - I realise it is slightly beyond the narrow scope of this PR, but I think for `str` we really should ensure the latitude carries a sign! I.e., in `sky_coordinate.to_string`, can we add `'alwayssign': True` to `dms` and `decimal` too? (I have no idea why this is not the default; pinging @eteq).

p.s. The travis failures are for unicode issues for python2. From a previous exchange I vaguely remember that one should not use `unicode_literals` any more, but it may also be as simple as adding a `str()` around the desired part in the assert statement.

**Comment by astrofrog** [comment]
Damn this Python 2 nonsense. How long are we going to have to support that archaic version? :stuck_out_tongue:

**Comment by eteq** [comment]
Hmm, there are three aspects of this that concern me:

First, I think the current implementation is problematic in how it deals with units - that is, it doesn't show them at all.  For `Time` this is a non-issue because there's enough context to make it clear what the units are... but that's not true for coordinates.  A simple fix might be to just add 'deg' (or 'd') to the end if we're going to actually require that it always be decimal degrees.

Second, I think it's also an issue to always require decimal degrees.  The following example illustrates the problem:
```
sc = SkyCoord(x=1*u.km,y=2*u.km,z=3*u.km,frame='itrs')
>>> str(sc)
'63.4349 53.3008'
```
This is happening because the ITRS frame defaults to a Cartesian representation, not Spherical, but ``to_string`` is making it into decimal degrees.  That certainly seems like unexpected behavior.  So instead I would think at least the units/representation should be the same as what ``repr`` gives.  (Note that this makes the first problem more pressing in that different outputs will have different units)

Third, I'm worried that this now drops *all* information about what the frame is.  As @taldcroft and @astrofrog said, there's precedence in `Time` in that the scale is dropped.  But I can't shake the feeling that this is much more important in coordinates, as the different components might mean something entirely different (whereas in `Time`, the different time scales are relatively subtle differences of a similar physical quantity).  A possible solution might be to have ``__str__`` give out something like 'ICRS: 1 2' or 'Galactic: 12.1 34.6' or similar... That's now getting pretty far afield of what ``to_string`` gives, though, so this point I can be persuaded against if the rest of you really aren't worried about this.

**Comment by astrofrog** [comment]
@eteq - I share some of those concerns. At the end of the day I think what I would like is to make sure that the output from tables looks sensible. The issue with having the frame in the line with the coordinates is that a table would then look like

```
          coords
--------------------------
   Galactic: 12.1 34.6
   Galactic: 12.2 34.6
   Galactic: 12.3 34.6
   Galactic: 12.4 34.6
   Galactic: 12.5 34.6
   Galactic: 12.6 34.6
```

Having said that I think that ``12.6 34.6`` is maybe too little so there is a balance to achieve here.

**Comment by mhvk** [comment]
@eteq - definitely some nails on the head, there! I do think the automatic conversion to a pair of angles in degrees is really incorrect, given your example. It is also inconsistent with what `Time` does: it doesn't just always convert to `jd`, but respect what the user put in. So, I think logically, we should keep cartesian if that was what was entered, and hexagesimal if that was what was put in, etc. I'd also advocate that if just numbers are presented, the unit is kept.

That said, I'm much less worried about the system: I think it is for the user to know what system they used; again, there is precedence in `Time` in not automatically converting to anything different from what the user put in, but also not worrying about indicating the system:
```
In [195]: str(Time.now())
Out[195]: '2015-01-21 21:10:30.074440'
In [198]: str(Time(Time.now(), format='mjd'))
Out[198]: '57043.88259317037'

In [199]: str(Time(Time.now(), format='cxcsec'))
Out[199]: '538261928.19372'
```

Overall, I fear that @taldcroft is right, and `SkyCoord` will have to keep this state (or perhaps `Angle` has to...). I'd be inclined to postpone this beyond 1.0 for that reason, though can see arguments as well for it being at least a little better with the PR (though, please, at least add the sign to latitude!).

**Comment by mhvk** [comment]
p.s. In the table context, I'd hope both system and units could be listed as the "unit".

**Comment by astrofrog** [comment]
I've removed the milestone since this is going to require more discussion. And I agree with @mhvk that whatever happens, the latitude should include a sign ;)

**Comment by mhvk** [comment]
Closing this in favour of new issue #6617

**Pull request closed without merging**